Automattic/jetpack
# Fix/one time payment as option for tier

Fixes 185-gh-automattic/gold / 8576-gh-automattic/jpop-issues

## Proposed changes:

* Filter out products with `one-time` payment intervals from the list of paid newsletter plans.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:







* Sandbox the public api and subscriptions.
* Configure the sandbox jetpack constant and enable global-http
* Create a new JN site but do not activate this feature branch yet - remain on trunk so that the problem can be shown.
* Connect Jetpack and Stripe
* Setup a monthly subscription (not a newsletter tier)
* Setup a one-time payment interval plan
* On your sandbox, open up wp shell get the list of products with `print_r(Memberships_Product::get_product_list(<your blog id>)` replacing `<your blog id>` with your blog id. Find the `connected_account_product_id` of the one-time payment plan you setup.
* Modify the plan to simulate an migrated plan with this issue. `Memberships_Product::get_from_post(<your blog id>,<your product id>)->update(['subscribe_as_site_subscriber'=>1, 'interval' => 'one-time']);` replacing your blog id and your product id accordingly.
* Setup a newsletter tier
* Create a post and set its access level to paid subscribers only
* Delete the newsletter tier
* In a new private window, navigate to the paid-subscribers-only post
* Enter an email address of a non-subscriber and click subscribe.
* Here, you should see the problem: there is only one option, but it's disabled. If you click pay, you'll see a "are you lost" message. Now, let's fix the issue.
* Patch your sandbox with D126371-code
* Activate this feature branch on the JN site
* In a new private window, navigate to the paid-subscribers-only post
* It should ask to confirm your email since no matching plans are available.
